src/Storages/PostgreSQL/PostgreSQLReplicationHandler.cpp

```diff
@@ -591,6 +642,88 @@ PostgreSQLTableStructurePtr PostgreSQLReplicationHandler::fetchTableStructure(
 }
 
 
+void PostgreSQLReplicationHandler::addTableToReplication(StorageMaterializedPostgreSQL * materialized_storage, const String & postgres_table_name)
+{
+    /// Note: we have to ensure that replication consumer task is stopped when we reload table, because otherwise
+    /// it can read wal beyond start lsn position (from which this table is being loaded), which will result in losing data.
+    consumer_task->deactivate();
```

Comment:
AFAIK It will not finish instantly after this call, task can be still in progress . Is it OK?